bridge-sdk/connectors/omni-connector/src/omni_connector.rs

```diff
@@ -427,21 +462,62 @@ impl OmniConnector {
         sender_id: Option<AccountId>,
         solana_token: Pubkey, // TODO: retrieve from near contract
     ) -> Result<Signature> {
-        let solana_bridge_client = self.solana_bridge_client()?;
         let near_bridge_client = self.near_bridge_client()?;
 
         let transfer_log = near_bridge_client
             .extract_transfer_log(near_tx_hash, sender_id, "SignTransferEvent")
             .await
             .map_err(|_| BridgeSdkError::UnknownError)?;
 
+        self.solana_finalize_transfer_with_event(
+            serde_json::from_str(&transfer_log).map_err(|_| BridgeSdkError::UnknownError)?,
+            solana_token,
+        )
+        .await
+        // TODO: This will silence a real error, so it must be fixed during `BridgeSdkError` refactoring
+        .map_err(|_| BridgeSdkError::UnknownError)
+    }
+
+    pub async fn solana_finalize_transfer_with_event(
+        &self,
+        event: Nep141LockerEvent,
+        solana_token: Pubkey, // TODO: retrieve from near contract
+    ) -> Result<Signature> {
+        let Nep141LockerEvent::SignTransferEvent {
+            message_payload,
+            signature,
+        } = event
+        else {
+            return Err(BridgeSdkError::UnknownError);
+        };
+
+        let solana_bridge_client = self.solana_bridge_client()?;
+
+        let mut signature = signature.to_bytes();
+        signature[64] -= 27;
+
+        let payload = FinalizeDepositData {
```

Comment:
The part where we transform omni_types into solana-specific structures part of the solana_bridge_client IMO

Reply:
Not sure if I understand this. Are you proposing moving everything starting from a `let mut signature` to the separate function that will be located in solana_bridge_client?